unison-runtime/src/Unison/Runtime/Builtin.hs

```diff
@@ -1720,59 +1646,63 @@ boxBoxBoxToBool =
 -- Works for an type that's packed into a word, just
 -- pass `wordDirect Ty.natRef`, `wordDirect Ty.floatRef`
 -- etc
+--
+-- TODO: Do we still need this?
 wordDirect :: Reference -> ForeignOp
-wordDirect wordType instr =
+wordDirect _wordType instr =
   ([BX],)
-    . TAbss [b1]
-    . unbox b1 wordType ub1
+    . TAbss [ub1]
     $ TFOp instr [ub1]
   where
-    (b1, ub1) = fresh
+    ub1 = fresh1
 
 -- Nat -> Bool
+--
+-- TODO: Do we still need this?
 boxWordToBool :: Reference -> ForeignOp
```

Comment:
I agree with these todos. It seems like most of the number-specific wrappers can just be replaced with more general wrappers.